src/devices/BuildHat/Brick.cs

```diff
@@ -0,0 +1,1564 @@
+        {
+            if (seconds <= 0)
+            {
+                // No need to move!
+                return;
+            }
+
+            if (!IsMotor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not a motor connected");
+            }
+
+            if (!IsActiveSensor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not an active motor connected");
+            }
+
+            if (speed == 0)
+            {
+                throw new ArgumentException("Speed can't be 0");
+            }
+
+            speed = speed < -100 ? -100 : speed > 100 ? 100 : speed;
+            ActiveMotor motor = (ActiveMotor)_elements[(byte)port];
+            // Set continuous reading
+            SelectCombiModesAndRead(port, new int[] { 1, 2, 3 }, false);
+            PortWrite($"port {(byte)port} ; pid {(byte)port} 0 0 s1 1 0 0.003 0.01 0 100; set pulse {speed} 0.0 {seconds.ToString(CultureInfo.InvariantCulture)} 0\r");
+            if (blocking)
+            {
+                Thread.Sleep((int)(seconds * 1000));
+            }
+        }
+
+        /// <summary>
+        /// Run the motor to an absolute position.
+        /// </summary>
+        /// <param name="port">The port.</param>
+        /// <param name="targetPosition">The target angle from -180 to +179.</param>
+        /// <param name="way">The way to go to the position.</param>
+        /// <param name="speed">The speed from - 100 to 100.</param>
+        /// <param name="blocking">True to block the function and wait for the execution.</param>
+        /// <exception cref="ArgumentException">Not a motor or not an active motor.</exception>
+        public void MoveMotorToAbsolutePosition(SensorPort port, int targetPosition, PositionWay way, int speed, bool blocking = false)
+        {
+            if (!IsMotor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not a motor connected");
+            }
+
+            if (!IsActiveSensor(_sensorType[(byte)port]))
+            {
+                throw new ArgumentException("Not an active motor connected");
+            }
+
+            if ((targetPosition < -180) || (targetPosition > 179))
+            {
+                throw new ArgumentException("Target position can only be between -180 and +179");
+            }
+
+            if (speed == 0)
+            {
+                throw new ArgumentException("Speed can't be 0");
+            }
+
+            speed = speed < -100 ? -100 : speed > 100 ? 100 : speed;
+            ActiveMotor motor = (ActiveMotor)_elements[(byte)port];
+            // We need both the relative and absolute position
+            int actualPosition = motor.Position;
+            int actualAbsolutePosition = motor.AbsolutePosition;
+            double actualPositionDouble = actualPosition / 360.0;
+            int difference = (targetPosition - actualAbsolutePosition + 180) % 360 - 180;
```

Comment:
You should write some unit tests for this function. I'm not sure the values are always in range after the calculations.

Reply:
And it was useful to ad. Found out one mistake. Thanks!